Unchanged example app crashes on startup on real Android device but not emulator

On my Android emulator the example app works perfectly, but if I try to start it on my Samsung Galaxy Note 5 (Android Version 7.0) then it crashes on initialization with the error, "flutter_local_notifications_example" has stopped. I haven't made any changes to the example app.

************************    This is the error on the actual Samsung Note 5 device: ************************

Launching lib\main.dart on SM N920T in debug mode...
Running Gradle task 'assembleDebug'...
√  Built build\app\outputs\flutter-apk\app-debug.apk.
W/FlutterActivityAndFragmentDelegate(19546): A splash screen was provided to Flutter, but this is deprecated. See flutter.dev/go/android-splash-migration for migration steps.
Debug service listening on ws://127.0.0.1:49933/8wH26ZocRDY=/ws
Syncing files to device SM N920T...
E/AndroidRuntime(19546): FATAL EXCEPTION: main
E/AndroidRuntime(19546): Process: com.dexterous.flutter_local_notifications_example, PID: 19546
E/AndroidRuntime(19546): java.lang.NoClassDefFoundError: Failed resolution of: Ljava/time/ZoneId;
E/AndroidRuntime(19546): 	at com.whelksoft.flutter_native_timezone.FlutterNativeTimezonePlugin.onMethodCall(FlutterNativeTimezonePlugin.kt:37)
E/AndroidRuntime(19546): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:233)
E/AndroidRuntime(19546): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:84)
E/AndroidRuntime(19546): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:865)
E/AndroidRuntime(19546): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/AndroidRuntime(19546): 	at android.os.MessageQueue.next(MessageQueue.java:323)
E/AndroidRuntime(19546): 	at android.os.Looper.loop(Looper.java:136)
E/AndroidRuntime(19546): 	at android.app.ActivityThread.main(ActivityThread.java:6682)
E/AndroidRuntime(19546): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(19546): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1534)
E/AndroidRuntime(19546): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1424)
E/AndroidRuntime(19546): Caused by: java.lang.ClassNotFoundException: Didn't find class "java.time.ZoneId" on path: DexPathList[[zip file "/data/app/com.dexterous.flutter_local_notifications_example-2/base.apk"],nativeLibraryDirectories=[/data/app/com.dexterous.flutter_local_notifications_example-2/lib/arm64, /data/app/com.dexterous.flutter_local_notifications_example-2/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64]]
E/AndroidRuntime(19546): 	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
E/AndroidRuntime(19546): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:380)
E/AndroidRuntime(19546): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
E/AndroidRuntime(19546): 	... 11 more

************* This is what is expected and is shown on the console when running on an Android emulator: ************

Launching lib\main.dart on sdk gphone x86 arm in debug mode...
Running Gradle task 'assembleDebug'...
√  Built build\app\outputs\flutter-apk\app-debug.apk.
Installing build\app\outputs\flutter-apk\app.apk...
W/FlutterActivityAndFragmentDelegate( 6296): A splash screen was provided to Flutter, but this is deprecated. See flutter.dev/go/android-splash-migration for migration steps.
Debug service listening on ws://127.0.0.1:59950/Itb8DzjTSTo=/ws
Syncing files to device sdk gphone x86 arm...

--- a/flutter_local_notifications/CHANGELOG.md
+++ b/flutter_local_notifications/CHANGELOG.md
@@ -1,7 +1,9 @@
 # [9.0.2]
 
 * [Android] Fixed issue [1357](https://github.com/MaikuB/flutter_local_notifications/issues/1357) where some details of a notification with formatted content weren't being returned
+* Bumped dependencies used by example app
 * Fixed grammar issue in readme in the `Scheduled Android notifications` section. Thanks to [Yousef Akiba](https://github.com/yousefakiba) for the PR
+* Updated example app and readme code around importing the `timezone` dependency so it uses the `all` variant of the IANA database that contains all timezones including those that are deprecated or may link to other timezones
 
 # [9.0.1]
 

--- a/flutter_local_notifications/README.md
+++ b/flutter_local_notifications/README.md
@@ -419,7 +419,7 @@ Usage of the `timezone` package requires initialisation that is covered in the p
 Import the `timezone` package
 
 ```dart
-import 'package:timezone/data/latest.dart' as tz;
+import 'package:timezone/data/latest_all.dart' as tz;
 import 'package:timezone/timezone.dart' as tz;
 ```
 

--- a/flutter_local_notifications/example/lib/main.dart
+++ b/flutter_local_notifications/example/lib/main.dart
@@ -15,7 +15,7 @@ import 'package:http/http.dart' as http;
 import 'package:image/image.dart' as image;
 import 'package:path_provider/path_provider.dart';
 import 'package:rxdart/subjects.dart';
-import 'package:timezone/data/latest.dart' as tz;
+import 'package:timezone/data/latest_all.dart' as tz;
 import 'package:timezone/timezone.dart' as tz;
 
 final FlutterLocalNotificationsPlugin flutterLocalNotificationsPlugin =

--- a/flutter_local_notifications/example/pubspec.yaml
+++ b/flutter_local_notifications/example/pubspec.yaml
@@ -3,18 +3,18 @@ description: Demonstrates how to use the flutter_local_notifications plugin.
 publish_to: none
 
 dependencies:
+  cupertino_icons: ^1.0.2
+  device_info: ^2.0.2
   flutter:
     sdk: flutter
   flutter_local_notifications:
     path: ../
-  cupertino_icons: ^1.0.2
-  device_info: ^2.0.0
-  flutter_native_timezone: ^1.0.10
-  http: ^0.13.0
+  flutter_native_timezone: ^2.0.0
+  http: ^0.13.4
+  image: ^3.0.8
   path_provider: ^2.0.0
-  rxdart: ^0.26.0-nullsafety.1
+  rxdart: ^0.27.2
   shared_preferences: ^2.0.1
-  image: ^3.0.2
 
 dev_dependencies:
   flutter_driver:
@@ -31,5 +31,5 @@ flutter:
     - sound/
 
 environment:
-  sdk: ">=2.12.0-0 <3.0.0"
-  flutter: ">=1.26.0-0" # allows for using integration_test from SDK
+  sdk: '>=2.12.0-0 <3.0.0'
+  flutter: '>=1.26.0-0' # allows for using integration_test from SDK

